baybe/surrogates/gaussian_process/presets/default.py

```diff
@@ -0,0 +1,110 @@
+"""Default preset for Gaussian process surrogates."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from attrs import define
+
+from baybe.kernels.basic import MaternKernel, ScaleKernel
+from baybe.priors.basic import GammaPrior
+from baybe.surrogates.gaussian_process.kernel_factory import KernelFactory
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+    from baybe.kernels.base import Kernel
+    from baybe.searchspace.core import SearchSpace
+
+
+@define
+class DefaultKernelFactory(KernelFactory):
+    """A factory providing the default kernel for the Gaussian process surrogate.
+
+    The logic is adapted from EDBO (Experimental Design via Bayesian Optimization).
+
+    References:
+    *   https://github.com/b-shields/edbo
+    *   https://doi.org/10.1038/s41586-021-03213-y
+    """
+
+    def __call__(  # noqa: D102
+        self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
+    ) -> Kernel:
+        # See base class.
+
+        mordred = (searchspace.contains_mordred or searchspace.contains_rdkit) and (
+            train_x.shape[-1] >= 50
+        )
+
+        # low D priors
+        if train_x.shape[-1] < 10:  # <-- different condition compared to EDBO
+            lengthscale_prior = GammaPrior(1.2, 1.1)
+            lengthscale_initial_value = 0.2
+            outputscale_prior = GammaPrior(5.0, 0.5)
+            outputscale_initial_value = 8.0
+
+        # DFT optimized priors
+        elif mordred and train_x.shape[-1] < 100:
+            lengthscale_prior = GammaPrior(2.0, 0.2)
+            lengthscale_initial_value = 5.0
+            outputscale_prior = GammaPrior(5.0, 0.5)
+            outputscale_initial_value = 8.0
+
+        # Mordred optimized priors
+        elif mordred:
+            lengthscale_prior = GammaPrior(2.0, 0.1)
+            lengthscale_initial_value = 10.0
+            outputscale_prior = GammaPrior(2.0, 0.1)
+            outputscale_initial_value = 10.0
+
+        # OHE optimized priors
+        else:
+            lengthscale_prior = GammaPrior(3.0, 1.0)
+            lengthscale_initial_value = 2.0
+            outputscale_prior = GammaPrior(5.0, 0.2)
+            outputscale_initial_value = 20.0
+
+        return ScaleKernel(
+            MaternKernel(
+                nu=2.5,
+                lengthscale_prior=lengthscale_prior,
+                lengthscale_initial_value=lengthscale_initial_value,
+            ),
+            outputscale_prior=outputscale_prior,
+            outputscale_initial_value=outputscale_initial_value,
+        )
+
+
+def _default_noise_factory(
```

Comment:
Because of the TODO noted in the function body: `TODO: Replace this function with a proper likelihood factory`.

So far, I've only refactored what was already available in the GP class, namely the kernel. That is, I made the kernel fully configurable for the user (circumventing the problems you also noticed when adding the `kernel` attribute, which were that you could nowhere specify the prior defaults etc. With the new factory approach, this is now possible). However, the likelihood hasn't even been configurable at all, so far. Therefore, I only extracted the settings into this function in order to separate it from the already refactored code and as a preparation for the next extension of making it controllable. We can do it now either in this PR as well (provided we all agree on my proposed factory approach, then I can replicate the same mechanism for the likelihood) or we do it in a next PR. 

Let me know what you think/prefer, @AVHopp, @Scienfitz

Reply:
I would prefer if the likelihood gets its proper factory then too already in this PR unless its a lot of work/code
This PR anyway is not strictly a refactoring anymore

Reply:
Not too much work, I think. I can add the necessary commits on top once we've all agreed on the final mechanism (i.e. resolved all threads and comments). Can then ping you and you can decide if you want to do another round of review for the remaining commits or if we first merge without and have an immediate follow-up PR ready 👍🏼 Will keep this thead open until then

Reply:
Have decided to postpone it to the next PR (which I'll immediately implement) since content will otherwise become too large